frame/election-providers/src/two_phase/mod.rs

```diff
@@ -0,0 +1,1622 @@
+	pub enum Event<T: Config>
+	where
+		ExtendedBalance: From<InnerOf<CompactAccuracyOf<T>>>,
+		ExtendedBalance: From<InnerOf<OnChainAccuracyOf<T>>>,
+	{
+		/// A solution was stored with the given compute.
+		///
+		/// If the solution is signed, this means that it hasn't yet been processed. If the solution
+		/// is unsigned, this means that it has also been processed.
+		SolutionStored(ElectionCompute),
+		/// The election has been finalized, with `Some` of the given computation, or else if the
+		/// election failed, `None`.
+		ElectionFinalized(Option<ElectionCompute>),
+		/// An account has been rewarded for their signed submission being finalized.
+		Rewarded(<T as frame_system::Config>::AccountId),
+		/// An account has been slashed for submitting an invalid signed submission.
+		Slashed(<T as frame_system::Config>::AccountId),
+		/// The signed phase of the given round has started.
+		SignedPhaseStarted(u32),
+		/// The unsigned phase of the given round has started.
+		UnsignedPhaseStarted(u32),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Submission was too early.
+		EarlySubmission,
+		/// Submission was too weak, score-wise.
+		WeakSubmission,
+		/// The queue was full, and the solution was not better than any of the existing ones.
+		QueueFull,
+		/// The origin failed to pay the deposit.
+		CannotPayDeposit,
+		/// WitnessData is invalid.
+		InvalidWitness,
+	}
+
+	#[pallet::origin]
+	pub struct Origin<T>(PhantomData<T>);
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T>
+	where
+		ExtendedBalance: From<InnerOf<CompactAccuracyOf<T>>>,
+		ExtendedBalance: From<InnerOf<OnChainAccuracyOf<T>>>,
+	{
+		type Call = Call<T>;
+		fn validate_unsigned(source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			if let Call::submit_unsigned(solution, _) = call {
+				// discard solution not coming from the local OCW.
+				match source {
+					TransactionSource::Local | TransactionSource::InBlock => { /* allowed */ }
+					_ => {
+						return InvalidTransaction::Call.into();
+					}
+				}
+
+				let _ = Self::unsigned_pre_dispatch_checks(solution)
+					.map_err(|err| {
+						log!(error, "unsigned transaction validation failed due to {:?}", err);
+						err
+					})
+					.map_err(dispatch_error_to_invalid)
+					.map(Into::into)?;
+
+				ValidTransaction::with_tag_prefix("OffchainElection")
+					// The higher the score[0], the better a solution is.
+					.priority(
+						T::UnsignedPriority::get()
+							.saturating_add(solution.score[0].saturated_into()),
+					)
+					// used to deduplicate unsigned solutions: each validator should produce one
+					// solution per round at most, and solutions are not propagate.
+					.and_provides(solution.round)
+					// transaction should stay in the pool for the duration of the unsigned phase.
+					.longevity(
+						TryInto::<u64>::try_into(T::UnsignedPhase::get())
+							.unwrap_or(crate::two_phase::unsigned::DEFAULT_LONGEVITY),
```

Comment:
I think you can use saturated_into
```suggestion
						T::UnsignedPhase::get().saturated_into()
```